uvicorn/protocols/http/h11_impl.py

```diff
@@ -155,6 +155,23 @@ def _unset_keepalive_if_required(self) -> None:
             self.timeout_keep_alive_task.cancel()
             self.timeout_keep_alive_task = None
 
+    def _should_upgrade(
+        self, headers: List[Tuple[bytes, bytes]], event: H11Event
+    ) -> bool:
+        connection = []
+        upgrade = None
+        for name, value in headers:
+            if name == b"connection":
+                connection = [token.lower().strip() for token in value.split(b",")]
+            if name == b"upgrade":
+                upgrade = value.lower().strip()
```

Comment:
Is there a reason for stripping?

```suggestion
                upgrade = value.lower()
```

Reply:
No specific reason, kept it for symmetry reasons and because I generally like to sanitize inputs. Removed it